src/test/kotlin/org/rust/lang/core/resolve/ref/RsLiteralWebReferenceProviderTest.kt

```diff
@@ -48,6 +48,14 @@ class RsLiteralWebReferenceProviderTest : RsTestBase() {
         }
     """, "http://localhost:8080", "https://github.com/foo/bar")
 
+    fun `test for invalid rust issue row`() = checkUrlReferences("""
```

Comment:
It would be nice to have positive test as well

Reply:
I didn't find an example of opening a standard library file. Did I miss something?

Reply:
I haven't looked at the `StdlibProjectDescriptor` code in detail, I was looking for a syntax like `// - ` 😂

Reply:
Oh, `StdlibProjectDescriptor` haven't existed yet. I suggest you to add it to have test package with `PackageOrigin.STDLIB`

> I was looking for a syntax like // -  😂

Yep, I thought about it as well, but relying on content of real stdlib may produce broken tests after updating rust version that we use in tests

Reply:
Does this warning need fixing?

<details>
  <pre>
    <code>
[   2014]   WARN - #c.i.o.v.i.l.WatchRootsManager - invalid watch root
java.nio.file.InvalidPathException: Watch roots should be absolute: /src
	at com.intellij.openapi.vfs.impl.local.WatchRootsManager.prepareWatchRoot(WatchRootsManager.java:222)
	at com.intellij.openapi.vfs.impl.local.WatchRootsManager.updateWatchRoots(WatchRootsManager.java:172)
	at com.intellij.openapi.vfs.impl.local.WatchRootsManager.replaceWatchedRoots(WatchRootsManager.java:68)
	at com.intellij.openapi.vfs.impl.local.LocalFileSystemImpl.replaceWatchedRoots(LocalFileSystemImpl.java:173)
	at com.intellij.openapi.roots.impl.ProjectRootManagerComponent.postCollect(ProjectRootManagerComponent.kt:215)
	at com.intellij.openapi.roots.impl.ProjectRootManagerComponent.addRootsToWatch(ProjectRootManagerComponent.kt:194)
	at com.intellij.openapi.roots.impl.ProjectRootManagerComponent.fireRootsChangedEvent(ProjectRootManagerComponent.kt:252)
	at com.intellij.openapi.roots.impl.ProjectRootManagerImpl.fireRootsChanged(ProjectRootManagerImpl.java:483)
	at com.intellij.openapi.roots.impl.ProjectRootManagerImpl$1.fireRootsChanged(ProjectRootManagerImpl.java:136)
	at com.intellij.openapi.roots.impl.ProjectRootManagerImpl$1.fireRootsChanged(ProjectRootManagerImpl.java:133)
	at com.intellij.openapi.roots.impl.ProjectRootManagerImpl$BatchSession.rootsChanged(ProjectRootManagerImpl.java:107)
	at com.intellij.workspaceModel.ide.impl.legacyBridge.project.ProjectRootsChangeListener.changed(ProjectRootsChangeListener.kt:42)
	at com.intellij.workspaceModel.ide.impl.legacyBridge.module.ModuleManagerComponentBridge$2.changed(ModuleManagerComponentBridge.kt:148)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeMethod(MessageBusImpl.kt:655)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeListener(MessageBusImpl.kt:625)
	at com.intellij.util.messages.impl.MessageBusImplKt.executeOrAddToQueue(MessageBusImpl.kt:466)
	at com.intellij.util.messages.impl.MessagePublisher.publish$intellij_platform_core(MessageBusImpl.kt:444)
	at com.intellij.util.messages.impl.MessagePublisher.invoke(MessageBusImpl.kt:421)
	at jdk.proxy3/jdk.proxy3.$Proxy32.changed(Unknown Source)
	at com.intellij.workspaceModel.ide.impl.WorkspaceModelImpl$onChanged$2.invoke(WorkspaceModelImpl.kt:234)
	at com.intellij.workspaceModel.ide.impl.WorkspaceModelImpl$onChanged$2.invoke(WorkspaceModelImpl.kt:233)
	at com.intellij.workspaceModel.ide.impl.WorkspaceModelImpl.logErrorOnEventHandling(WorkspaceModelImpl.kt:255)
	at com.intellij.workspaceModel.ide.impl.WorkspaceModelImpl.onChanged(WorkspaceModelImpl.kt:233)
	at com.intellij.workspaceModel.ide.impl.WorkspaceModelImpl.access$onChanged(WorkspaceModelImpl.kt:29)
	at com.intellij.workspaceModel.ide.impl.WorkspaceModelImpl$updateProjectModel$generalTime$1$7.invoke(WorkspaceModelImpl.kt:130)
	at com.intellij.workspaceModel.ide.impl.WorkspaceModelImpl$updateProjectModel$generalTime$1$7.invoke(WorkspaceModelImpl.kt:130)
	at com.intellij.workspaceModel.storage.impl.VersionedEntityStorageImpl.replace(VersionedEntityStorageImpl.kt:186)
	at com.intellij.workspaceModel.ide.impl.WorkspaceModelImpl.updateProjectModel(WorkspaceModelImpl.kt:130)
	at com.intellij.workspaceModel.ide.impl.legacyBridge.module.roots.ModifiableRootModelBridgeImpl.commit(ModifiableRootModelBridgeImpl.kt:560)
	at com.intellij.openapi.application.WriteAction.lambda$run$1(WriteAction.java:86)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteActionWithClass(ApplicationImpl.java:943)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:969)
	at com.intellij.openapi.application.WriteAction.run(WriteAction.java:85)
	at com.intellij.openapi.roots.ModuleRootModificationUtil.lambda$modifyModel$12(ModuleRootModificationUtil.java:168)
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:408)
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:441)
	at com.intellij.openapi.roots.ModuleRootModificationUtil.modifyModel(ModuleRootModificationUtil.java:166)
	at com.intellij.openapi.roots.ModuleRootModificationUtil.updateModel(ModuleRootModificationUtil.java:156)
	at com.intellij.testFramework.LightProjectDescriptor.createContentEntry(LightProjectDescriptor.java:145)
	at com.intellij.testFramework.LightProjectDescriptor.lambda$setUpProject$0(LightProjectDescriptor.java:46)
	at com.intellij.openapi.application.WriteAction.lambda$run$1(WriteAction.java:86)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteActionWithClass(ApplicationImpl.java:943)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:969)
	at com.intellij.openapi.application.WriteAction.run(WriteAction.java:85)
	at com.intellij.testFramework.LightProjectDescriptor.setUpProject(LightProjectDescriptor.java:40)
	at com.intellij.testFramework.LightPlatformTestCase.initProject(LightPlatformTestCase.java:192)
	at com.intellij.testFramework.LightPlatformTestCase.lambda$doSetup$2(LightPlatformTestCase.java:273)
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:408)
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:441)
	at com.intellij.testFramework.LightPlatformTestCase.doSetup(LightPlatformTestCase.java:264)
	at com.intellij.testFramework.fixtures.impl.LightIdeaTestFixtureImpl.setUp(LightIdeaTestFixtureImpl.java:41)
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.lambda$setUp$32(CodeInsightTestFixtureImpl.java:1302)
	at com.intellij.testFramework.EdtTestUtil.runInEdtAndWait(EdtTestUtil.java:33)
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.setUp(CodeInsightTestFixtureImpl.java:1301)
	at com.intellij.testFramework.fixtures.BasePlatformTestCase.setUp(BasePlatformTestCase.java:48)
	at org.rust.RsTestBase.setUp(RsTestBase.kt:61)
	at com.intellij.testFramework.UsefulTestCase.invokeSetUp(UsefulTestCase.java:431)
	at com.intellij.testFramework.UsefulTestCase.defaultRunBare(UsefulTestCase.java:423)
	at com.intellij.testFramework.UsefulTestCase.lambda$runBare$12(UsefulTestCase.java:492)
	at com.intellij.testFramework.EdtTestUtil.lambda$runInEdtAndWait$1(EdtTestUtil.java:40)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:308)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:771)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:716)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:741)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:389)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
    </code>
  </pre>
</details>

Reply:
Seems suspicious
Maybe you push code which produces such exception? Otherwise it's hard to say why it happens

Reply:
> Seems suspicious Maybe you push code which produces such exception? Otherwise it's hard to say why it happens

```kotlin
return super.testCargoPackage(contentRoot, name).copy(origin = PackageOrigin.STDLIB)
```